lib/rules/no-constant-condition.js

```diff
@@ -32,17 +34,20 @@ module.exports = function(context) {
                 return true;
 
             case "UnaryExpression":
-                return isConstant(node.argument);
+                return (node.operator === "typeof" && inBooleanPosition) ||
+                    isConstant(node.argument, true);
 
             case "BinaryExpression":
             case "LogicalExpression":
-                return isConstant(node.left) && isConstant(node.right) && node.operator !== "in";
+                return isConstant(node.left, false) &&
+                        isConstant(node.right, false) &&
+                        node.operator !== "in";
 
             case "AssignmentExpression":
-                return (node.operator === "=") && isConstant(node.right);
+                return (node.operator === "=") && isConstant(node.right, true);
```

Comment:
This should pass `inBooleanPosition` through, not `true`. You should have tests for this.